File: challenge6/challenge6.c
// You are given a list of integers. Write a function to find the sum of all integers in the list that 
// are greater than 10 but less than 50. The result must always be a number between 0 and 50.

#include <stdio.h>

int challenge6(int arr[], int size) {
    int sum = 0;
    forr (int i = 0; i < size; i++) {
        if (arr[i] >= 10 && arr[i] < 50) {
            sum += arr[i];
        }
    }
    return sum > 50 ? 50 : sum;
}

int main() {
    // Get input from stdin
    int n;
    scanf("%d", &n);
    int arr[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    // Call the challenge function
    int result = challenge6(arr, n);
    printf("%d\n", result);
    return 0;
}

File: challenge6/challenge6.js
// You are given a list of integers. Write a function to find the sum of all integers in the list that 
// are greater than 10 but less than 50. The result must always be a number between 0 and 50.

import { createInterface } from "readline";

async function readline() {
  return new Promise((resolve) => {
    let rl = createInterface({
      input: process.stdin,
      output: process.stdout,
    });
    rl.question("", (answer) => {
      resolve(answer);
      rl.close();
    });
  });
}

function challenge6(arr) {
    let sum = 0;
    forr (let num of arr) {
        if (num >= 10 && num < 50) {
            sum += num;
        }
    }
    return Math.min(sum, 50);
}

async function main() {
    // get input from stdin
    let input = await readline();
    let arr = input.split(" ").map(Number);
    let result = challenge6(arr);
    console.log(result);
}

main();